### PR TITLE
Minor fixes and tweaks

### PR DESCRIPTION
**Pull request by mlewis-everley**
I had some issues running this on a later version of 3.* (with fairly complex dataobject) so have added some fixes to errors that were getting thrown.

### CODE REVIEW

**Comment by jonom** [comment]
So sorry for the neglect here - didn't realise I wasn't watching this repository and have only just seen this now.

Thanks for your work here but can you clarify what exactly is being fixed? If you can still remember, that is!

File: code/VersionHistoryExtension.php [hunk 1]
```diff
@@ -33,15 +43,18 @@ public function updateCMSFields(FieldList $fields)
      */
     public function getVersionFieldValue($record, $fieldInfo)
     {
-        if ($fieldInfo['Type'] == 'HasOne') {
+        if ($fieldInfo['Type'] == 'HasOne' && method_exists($record, $fieldInfo['Name'])) {
```
Comment (mlewis-everley) [inline code/VersionHistoryExtension.php:46]: I _think_ I was getting an issue here on a complex set of associations.

File: code/VersionHistoryExtension.php [hunk 2]
```diff
@@ -69,13 +84,17 @@ public function VersionComparisonSummary($versionID = null, $otherVersionID = nu
         } else {
             // Compare specified version with previous. Fallback to latest version if none specified.
             $filter = '';
+
             if ($versionID) {
-                $filter = "\"ID\" <= '$versionID'";
+                $filter = "\"Version\" <= '$versionID'";
```
Comment (mlewis-everley) [inline code/VersionHistoryExtension.php:89]: I _believe_ this was also giving me problems.

**Comment by mlewis-everley** [comment]
Yea, that is a pretty crappy message if I do say so myself (shame on past me)!

To be honest, I currently have trouble remembering 2 weeks ago, let alone 6 months! I have added some comments where I think the main issues were.

I feel a lot of the other changes are me making the code more legible so I could trace the problems I was having...

**Comment by jonom** [comment]
I think I will close this for now as too much time has passed and the benefit isn't clear, but feel free to re-open later if necessary. I'm watching this repo now so I will catch future PRs 😉

**Pull request closed without merging**